agent/agent.go

```diff
@@ -1027,6 +1022,12 @@ func (a *Agent) finishTask(ctx context.Context, tc *taskContext, status string,
 	}
 	grip.Infof("Successfully sent final task status: '%s'.", detail.Status)
 
+	// Upsert check run only after updating task status.
```

Comment:
super nit, feel free to ignore lol. 
I think it might be a good idea to either not include the comment, or elaborate a bit on why we have to do it after updating the task status.